plugins/sip/lib/model/providers/pexip/kPexipHandler.php

```diff
@@ -37,7 +39,12 @@ public static function createCallObjects(LiveStreamEntry $dbLiveEntry, $pexipCon
 		{
 			$locationId = $pexipConfig[kPexipUtils::CONFIG_PRIMARY_LOCATION_ID];
 			$locationId = is_numeric($locationId) ? $locationId : null;
-			$primaryAdpId = self::addADP($dbLiveEntry, $roomId, $primaryUrl, 'Primary', $pexipConfig, $locationId);
+			$dualLiveEntryPrimaryUrl = null;
```

Comment:
this piece of code repeats at line 59.
$primaryAdpId =  getCreatedAdpId(sourceType,$dualLiveEntry, $pexipConfig,true);
$secondaryAdpId =  getCreatedAdpId(sourceType,$dualLiveEntry, $pexipConfig,false);

You can even throw on failure inside.